pymesos/process.py

```diff
@@ -143,14 +147,16 @@ def abort(self):
 
     def stop(self):
         Process.abort(self)
-        self.join()
-        for addr in self.conn_pool:
-            self.conn_pool[addr].close()
-        self.conn_pool.clear()
+        if threading.current_thread() != self.accept_t:
```

Comment:
actruallly i tend to use `abort` for this case, `stop` is a blocking call and needs to ensure all threads are stopped. official mesos scheduler is based on coroutines and we are using threads, thus it is too different to keep the same behavior. i need more time to check this issue.